### PR TITLE
Feat: flag off comment replies

### PR DESCRIPTION
**Pull request by neymarjimoh**
**Before submitting your PR for review**

- Run `npm run lint` to find errors in code syntax/format
- Run `npm run lint:fix` to fix all auto-fixable errors in source code and auto-format with prettier
- Ensure you fix any linting errors displayed after running any of the above commands

**What does this PR do?**
- This PR contains the functionality for the ability of a comment's reply to be flagged off by users

Describe (in detail) what the task completed in the pull request does as per the relevant task

**How should this be manually tested?**
- This can be tested by :
- Run `npm install` to install all dependencies
- Run `npm run lint:fix` to fix all linting issues
- Run `npm run startDev` to start it locally
- Go to the endpoint  `/comments/{commentId}/replies/{replyId}/flag` with a **PATCH** http verb

Give bullet point instructions on how to setup the code and test it manually (Assume no prior experience on JS or code)

**Any background context you want to provide?**

Any pertinent information that should be considered

**What is the link to the issue on Github?**

[Issue #76](https://github.com/microapidev/comment-microapi/issues/76)

**Questions:**

If something is unclear or you want some questions to be addressed by your peers, mention them here

### CODE REVIEW

**Comment by neymarjimoh** [comment]
@dave-ok  @piouson @oscar-ekeyekwu  can you please review

**Comment by neymarjimoh** [comment]
Ok

**Comment by neymarjimoh** [comment]
The docs have been updated, I'd use it

File: controller/repliesController.js [hunk 1]
```diff
@@ -102,7 +102,67 @@ const createReply = async (req, res, next) => {
...
+          `Reply with the ID ${replyId} doesn't exist or has been deleted`
+        )
+      );
+      return;
+    }
+
+    //flag comment reply by pushing ownerId into flags array
+    if (!reply.flags.includes(ownerId)) {
+      reply.flags.push(ownerId);
+    }
```
Comment (simply-alliv) [inline controller/repliesController.js:138]: You need to add an else statement that will also remove the ownerId from the flags array if it already exists.

Think of it as a toggle.

File: routes/replies.js [hunk 2]
```diff
@@ -3,4 +3,5 @@ const repliesController = require("../controller/repliesController");
 
 router.get("/", repliesController.getCommentReplies);
 router.post("/", repliesController.createReply);
+router.patch("/:");
```
Comment (simply-alliv) [inline routes/replies.js:6]: Use the correct endpoint for flagging replies

**Pull request merged**